back/src/bsds/indexation/bulkIndexBsds.ts

```diff
@@ -0,0 +1,364 @@
+import type { ApiResponse } from "@elastic/elasticsearch";
+import logger from "../../logging/logger";
+import { client, indexAllBsds, BsdIndex } from "../../common/elastic";
+import { BsdType } from "../../generated/graphql/types";
+
+type IndexElasticSearchOpts = {
+  index: BsdIndex;
+  force?: boolean;
+  useQueue?: boolean;
+};
+
+/**
+ * Separators constants
+ * If changed, breaks the script roll-over system
+ * BETTER NOT change in production
+ */
+export const INDEX_ALIAS_NAME_SEPARATOR = "_";
+export const INDEX_DATETIME_SEPARATOR = "===";
+
+/**
+ * Convert Date to readable index version name suffix
+ */
+const getIndexDateString = (dateStr?: string) => {
+  let dateObj;
+  if (dateStr) {
+    dateObj = new Date(dateStr);
+  } else {
+    dateObj = new Date();
+  }
+
+  return dateObj
+    .toJSON()
+    .replace(/\:/g, INDEX_DATETIME_SEPARATOR)
+    .toLowerCase();
+};
+
+/**
+ * Convert index name to the Date object of its creation time
+ */
+const getIndexDateFromName = (indexName: string): Date => {
+  // Datetime is the last fragment in the index name
+  const indexDateString = indexName.split(INDEX_ALIAS_NAME_SEPARATOR).pop();
+  return new Date(
+    indexDateString
+      .toUpperCase()
+      .replace(new RegExp(INDEX_DATETIME_SEPARATOR, "g"), ":")
+  );
+};
+
+/**
+ * Convert index name to the mappings version name
+ */
+const getIndexMappingsVersionFromName = (indexName: string): string =>
+  indexName.split(INDEX_ALIAS_NAME_SEPARATOR)[1];
+
+/**
+ * Build a codified index name
+ */
+const getIndexName = (index: BsdIndex, dateStr?: string): string =>
+  `${index.alias}${INDEX_ALIAS_NAME_SEPARATOR}${
+    index.mappings_version
+  }${INDEX_ALIAS_NAME_SEPARATOR}${
+    process.env.NODE_ENV ? process.env.NODE_ENV : "dev"
+  }${INDEX_ALIAS_NAME_SEPARATOR}${getIndexDateString(dateStr)}`;
```

Comment:
Nitpick, mais ca serait p-e plus lisible comme suit:
```suggestion
const getIndexName = (index: BsdIndex, dateStr?: string): string =>
  [index.alias, index.mappings_version, process.env.NODE_ENV ?? "dev", getIndexDateString(dateStr)].join(INDEX_ALIAS_NAME_SEPARATOR)
```